vault/resource_database_secret_backend_connection.go

```diff
@@ -1257,12 +1260,30 @@ func databaseSecretBackendConnectionRead(d *schema.ResourceData, meta interface{
 		roles = append(roles, role.(string))
 	}
 
-	d.Set("allowed_roles", roles)
-	d.Set("backend", backend)
-	d.Set("name", name)
-	d.Set("root_rotation_statements", resp.Data["root_credentials_rotate_statements"])
+	if err := d.Set("allowed_roles", roles); err != nil {
```

Comment:
When is it okay to skip this error check? I see that we've added them in here, but still have them left out on call within the switch case above.

Reply:
Good question, I would say never. We have been trying update unhandled errors when we see them. Perhaps we should do that across the entire code base...